File: app/src/main/java/com/makingiants/todayhistory/today/TodayView.java
package com.makingiants.todayhistory.today;

import com.makingiants.today.api.repository.history.pojo.Event;
import com.makingiants.todayhistory.base.BaseActivityView;
import java.util.List;

public abstract class TodayView extends BaseActivityView {
  public abstract void showEvents(List<Event> events);

  public abstract void hideEvents();

  public abstract void showEmptyViewProgress();

  public abstract void dismissEmptyViewProgress();

  public abstract void dismissReloadProgress();

  public abstract void showEmptyView();

  public abstract void hideEmptyView();

  public abstract void showReloadProgress();

  public abstract void showErrorView(String title, String message);

  public abstract void hideErrorView();

  public abstract void showErrorToast(String message);
}

File: app/src/main/kotlin/com/makingiants/todayhistory/today/TodayView.kt
package com.makingiants.todayhistory.today

import com.makingiants.today.api.repository.history.pojo.Event

interface TodayView {
    fun initViews()

    fun showEvents(events: List<Event>)

    fun hideEvents()

    fun showEmptyViewProgress()

    fun dismissEmptyViewProgress()

    fun dismissReloadProgress()

    fun showEmptyView()

    fun hideEmptyView()

    fun showReloadProgress()

    fun showErrorView(title: String, message: String)

    fun hideErrorView()

    fun showErrorToast(message: String)

    fun showErrorDialog(throwable: Throwable)
}